### ISSUE TITLE
A new crash

### ISSUE THREAD

**Issue by Cat5TV** [stujones11/minetest-3d_armor#61]
Hey stujones11,
Another server crashed cropped up in the past few days...

ServerError: Lua: Runtime error from mod '3d_armor' in callback on_dieplayer(): ...ie/.minetest/mods/minetest-3d_armor/3d_armor/api.lua:115: attempt to index field '?' (a nil value)

Cheers,
Robbie

**Comment by stujones11** [comment]
That code should never be reached in on_dieplayer() if you have disabled armor dropping. Any chance you can provide the full traceback? I can only guess that some other mod is calling set_player_armor and I've somehow broken the compatibility.

btw, this is really a duplicate of #60 so I will close that one now.

**Comment by stujones11** [comment]
Try as I might, I cannot seem to reproduce this by any means of dying. In will need to update my minetest installation to see if it could be an engine problem but I doubt that. I have added some additional validation code around the problem area that should, if nothing else, provide me with a little more insight.

**Comment by Cat5TV** [comment]
~/.minetest/mods/minetest-3d_armor$ cat armor.conf 
ARMOR_DROP = false

Because of this issue I have players emailing me to tell me they can't connect to the server (the server crashes with the above error the moment they try to connect).

**Comment by stujones11** [comment]
Have you updated the model since my last few commits. The logs should now provide a little more info now. Also, please try to capture the full traceback if it still causes a crash.

**Comment by Cat5TV** [comment]
Alright will do. I disabled the daily build yesterday just because we've had so many problems lately with instability... everyone's trying to keep up  :)  I'll update now and see what we see over the next few days.

Thanks!

**Comment by stujones11** [comment]
No problem, fbdc68df5153592b5d22720911a668ff6bcc771e should now hopefully fix the problem regardless of the actual cause.

**Comment by Cat5TV** [comment]
Thanks man. Had a long weekend here so I've been AFK, but just did a new pull and will see how it shapes up.

**Comment by stujones11** [comment]
You are very welcome, please let me know the outcome either way because I would like to get 0.5.0 released sometime over the coming weekend.

**Comment by Cat5TV** [comment]
Sure thing. I've updated all, including your mods and a new compile on minetest just to make sure everything is as current as can be.

If you need me, ping me here or pop on #ThePixelShadow on Freenode.

Cheers.

**Comment by Cat5TV** [comment]
Non-trivial, he marks it  ;)

This morning after yesterday's updates:

2016-05-26 08:53:27: ACTION[Server]: Junebug leaves game. List of players: 
2016-05-26 08:53:28: ERROR[Main]: ServerError: Lua: Runtime error from mod '3d_armor' in callback on_dieplayer(): .../.minetest/mods/minetest-3d_armor/multiskin/init.lua:53: attempt to index field '?' (a nil value)
2016-05-26 08:53:28: ERROR[Main]: stack traceback:
2016-05-26 08:53:28: ERROR[Main]: 	.../.minetest/mods/minetest-3d_armor/multiskin/init.lua:53: in function 'set_player_textures'
2016-05-26 08:53:28: ERROR[Main]: 	...ie/.minetest/mods/minetest-3d_armor/3d_armor/api.lua:127: in function 'set_player_armor'
2016-05-26 08:53:28: ERROR[Main]: 	...e/.minetest/mods/minetest-3d_armor/3d_armor/init.lua:284: in function '?'
2016-05-26 08:53:28: ERROR[Main]: 	...ome/robbie/minetest/bin/../builtin/game/register.lua:369: in function <...ome/robbie/minetest/bin/../builtin/game/register.lua:349>
Minetest Stopped. List of players:

**Comment by numberZero** [comment]
It seems that `player:get_player_name()` returned a literal "?" instead of `nil` or the player name. That looks very strange but may be caused by some another mod.

BTW, what do is expected to happen if there is a player named e.g. `get_player_skin`?

    multiskin[name][layer] = textures[layer] or multiskin[name][layer]

The server crashes, of course:

    2016-05-26 19:41:35: ACTION[Server]: get_player_skin [127.0.0.1] joins game. 
    2016-05-26 19:41:35: ACTION[Server]: get_player_skin joins game. List of players: set_player_skin get_player_skin
    ...
    2016-05-26 19:42:19: ACTION[Server]: set_player_textures [127.0.0.1] joins game. 
    2016-05-26 19:42:19: ACTION[Server]: set_player_textures joins game. List of players: set_player_skin set_player_textures
    2016-05-26 19:42:19: ERROR[Main]: ServerError: Lua: Runtime error from mod 'multiskin' in callback on_joinplayer(): ...layable-minetest/bin/../mods/3d_armor/multiskin/init.lua:84: attempt to call method 'get_player_skin' (a table value)
    2016-05-26 19:42:19: ERROR[Main]: stack traceback:
    2016-05-26 19:42:19: ERROR[Main]:       ...layable-minetest/bin/../mods/3d_armor/multiskin/init.lua:84: in function <...layable-minetest/bin/../mods/3d_armor/multiskin/init.lua:81>
    2016-05-26 19:42:19: ERROR[Main]:       ...etest/playable-minetest/bin/../builtin/game/register.lua:369: in function <...etest/playable-minetest/bin/../builtin/game/register.lua:349>
    2016-05-26 19:42:19: ACTION[Server]: set_player_skin leaves game. List of players: 
    2016-05-26 19:42:19: ACTION[Server]: set_player_textures leaves game. List of players:

**Comment by stujones11** [comment]
@Cat5TV  something definitely seems to be corrupting the player name. All I can do is add more checks to prevent a crash but the problem is likely to re-emerge somewhere else.

> BTW, what do is expected to happen if there is a player named e.g. get_player_skin?

That is a good point, well spotted. Will fix that asap.

**Comment by stujones11** [comment]
fb2ff3a217b0156dd412b9b5a38937e8ed0ae477 should now fix the potential issue reported by @numberZero and at least prevent the crash for re-occurring there.

**Comment by Cat5TV** [comment]
I've done a new pull. Will let you know the outcome.

**Comment by stujones11** [comment]
Thanks, though I would still like to know what other mods you have activated as something is very wrong when the api is returning garbage player names. It has to be a mod problem because I have updated my mt/mtg installation but am still unable to reproduce this error.

**Comment by Cat5TV** [comment]
https://minetest.tv/server-specs.php - the server we had crash the other day was the "Survival" server.

**Comment by stujones11** [comment]
Wow, that's a lot of mods. I am now wondering if it is _only_ your survival server that is crashing, that would narrow things down considerably.

**Comment by Cat5TV** [comment]
It's really tough to know because it's only when certain players login. That said, it has happened on both Survival and Xtreme.

**Comment by stujones11** [comment]
I wonder if it be something in the player name then, eg. do they contain any non-english characters?

**Comment by Cat5TV** [comment]
2016-05-26 08:53:27: ACTION[Server]: Junebug leaves game. List of players: 

Player name was Junebug.

**Comment by stujones11** [comment]
Yeah, it was in the logs, derp, been one of them days...I still don't understand how it could happen when a player leaves. I guess we'll just have to see were it goes from here.

**Comment by Raughnbo** [comment]
Two of the characters that this is happening to are mine. One of the character's name is Junebug the others name is Fred. Both players died while AFK and now everytime I try to log either character into the server the server resets. I have two other characters on the server that login to the server without issue. If you have any specific questions please ask and I will answer them to the best of my ability. Thank you.

**Comment by stujones11** [comment]
@Raughnbo Thank you for this information, it has given me a clue as to what might be happening. If I am correct then it is really an engine bug, although mods should try avoid server crash in any case.

**Comment by stujones11** [comment]
@Cat5TV any word on the status of this? I would really like to get 0.5.0 released but I want to be sure this has been resolved before I do.

**Comment by Cat5TV** [comment]
Checking the logs for you now. I have not noticed any crashes as of late, but didn't realize you had changed anything since the last comment.

**Comment by Cat5TV** [comment]
@stujones11 the last instance of the crash was on May 26th, as detailed above. It has not appeared in our logs since that day.

**Comment by stujones11** [comment]
Thank you for the update, I do hope that @Raughnbo was able to recover his characters.

**Comment by Cat5TV** [comment]
I have asked @Raughnbo to sign in ... we'll see.

**Comment by Cat5TV** [comment]
No good.

2016-06-01 20:49:28: ACTION[Server]: TSmith [67.140.38.52] joins game.
2016-06-01 20:49:28: ACTION[Server]: TSmith joins game. List of players: Geraths TSmith
2016-06-01 20:49:28: ACTION[Main]: IRC: Disconnected.
2016-06-01 20:49:29: ERROR[Main]: ServerError: Lua: Runtime error from mod '' in callback on_joinplayer(): ...e/.minetest/mods/minetest-3d_armor/3d_armor/init.lua:92: attempt to index field '?' (a$
2016-06-01 20:49:29: ERROR[Main]: stack traceback:
2016-06-01 20:49:29: ERROR[Main]:       ...e/.minetest/mods/minetest-3d_armor/3d_armor/init.lua:92: in function 'get_formspec'

**Comment by Raughnbo** [comment]
No Good here

**Comment by stujones11** [comment]
> No good. Different line of init.lua though... perhaps unrelated?

It almost definitely related but it really should not be possible for the player's armor.def entry to be `nil`. This can only mean that either `register_on_joinplayer` is somehow being skipped for that player or some other mod is errantly modifying the armor definition table.

e6da523b979adf281da5b536212d387043a5a3f2 should stop it crashing there again anyway.

**Comment by Cat5TV** [comment]
Just a crazy stab in the dark, but could it be related to this? [https://github.com/NathanSalapat/kickafk](https://github.com/NathanSalapat/kickafk)

I only wonder because @Raughnbo said above "Both players died while AFK and now everytime I try to log either character into the server the server resets." -- and this mod specifically deals with AFK players.

Will e6da523 be integrated into 0.5.0?

Thanks!!

**Comment by stujones11** [comment]
@Cat5TV I think you might be on the right lines there but I cannot see anything in Nathan's code that could possibly explain this.

> Will e6da523 be integrated into 0.5.0?

Did you mean current master? That commit is already part of 0.5.0-dev, so it will be part of the eventual release in some form or other.

**Comment by numberZero** [comment]
...`from mod '' in`... — no mod name
...`field '?' (a nil`... — literal “?”, again (unless Cat5TV removed actual player name...)

Certainly something messes up the API. That may be a rogue mod, a broken patch for the builtins. strange engine bug, etc.
@Cat5TV can you give me a tarball of mods installed on your server? I would try to investigate that further then. Also, do you know who else suffers from this bug? Maybe Zool, maybe someone else?

**Comment by stujones11** [comment]
> ...field '?' (a nil... — literal “?”, again (unless Cat5TV removed actual player name...)

afaik, the '?' is not literal here, it effectively means that the code tried to do `nil[index]`. The mystery is how it being deleted or not initialized. Thank you for your assistance, I hope you are able to reproduce this anomaly and find the cause.

**Comment by Cat5TV** [comment]
@numberZero happy to provide a tar. However, out of respect for contributors who would prefer their custom work not be shared publicly, I cannot upload it here. Can I email it to your yandex address? Again, simply to honor the wishes of those who have coded some of the "not public" mods, I'd ask that you kindly use it only for testing this bug, not share it, and remove it when you are done. Cool?

Thanks so much!

**Comment by numberZero** [comment]
@stujones11 sorry, I forgot that Lua is so weird sometimes
@Cat5TV received, will try to find out what exactly happens

**Comment by numberZero** [comment]
reproduced

**Comment by Cat5TV** [comment]
Thanks numberZero - so glad you were able to replicate the problem. Here's hoping you more experienced folk can figure this one out. I'm here if you need me to answer any questions or provide any more info.

**Comment by stujones11** [comment]
> Crash happens, but in different way

I would guess that is the same problem, only a different error since I prevented it from re-occurring there. It's great that you have been able to reproduce this, hopefully now we can find the root cause.

What I find strange is why nobody has reported any issue with the master branch. It would be interesting to know if that is effected or whether this only happens with 0.5.0-dev

**Comment by Cat5TV** [comment]
Agreed @stujones11 - although I'm sure some people would just disable bits and move on if it had problems... whereas I _really_ want these features for our players!  :)

It is important to note as well, that the applicable mods themselves on our server (eg., 3D Armor, Unified Inventory) are all up to date with the latest git pulls.

**Comment by Cat5TV** [comment]
Any updates? Server still crashing with this. Thanks!!

**Comment by Raughnbo** [comment]
Just checked, both Junebug and Fred are now able to log into the Pixel
Shadow Survival Server.

Thanks,

~Raughnbo

**Comment by Cat5TV** [comment]
@Raughnbo while I'm glad you're able to login, this is because I finally gave up on 3D_armor 0.5.0 and downgraded to 0.4.5. So you're able to login again because of a downgrade. Several features will be missing (eg., armor stands, shield frames), but at least it doesn't crash the server. Hopefully Stu can get this solved so when 0.5.0 comes out, we can upgrade.

**Comment by Raughnbo** [comment]
Thanks Robbie, I hope the problem gets solved too.

**Comment by stujones11** [comment]
@Cat5TV so you can confirm that is only happens with the 0.5.0-dev branch? I was hoping that @numberZero might have come back with a solution but since this information narrows things down a lot, I will take another look at it myself. btw, you should be able to use the armor and shield stands from the dev branch if you wanted, only clothing won't work without the multiskin mod.

**Comment by Cat5TV** [comment]
@stujones11 yay, let the complaints about clothing begin!  ;)  Yes, after downgrading to 0.4.5 the issue went away, or at least seemingly so (the players who couldn't login and would crash our server when they tried can now login just fine).

Sorry I'm not well versed at Lua, otherwise I'd love to be more help.

**Comment by stujones11** [comment]
I think I have finally figured this one out and it really comes down to an engine bug: https://github.com/minetest/minetest/issues/3565

When a player dies but does not respawn, the on_dieplayer callback is being re-called before any of the mod initialization, hence the nil values.  It only does not happen in 0.4.5 is because the old get_valid_player method checked for a valid detached inventory. If I am correct then you should be seeing lots of `3d_armor: Detached armor inventory is nil` messages in your logs.

I have submitted a patch to the 0.5.0-dev branch to make it behave the same as 0.4.5 under these circumstances, it is not really a fix but a work-around until root cause is fixed. Note that you will continue to receive the warning messages.

**Comment by Cat5TV** [comment]
@stujones11 yes, I ran a grep and in the course of 4 days we had 84 instances of that message in the logs.

Okay, I will reinstall 0.5.0-dev and hope all is well  :)  @Raughnbo can you let us know if there are any problems after the upgrade?

Thanks so much Stu. Keep up the great work, and let me know what I can do to help.

**Comment by stujones11** [comment]
If you could get someone to die but not respawn until the server is restarted, then you can be reasonably sure the crash is fixed, however, there may still be other problems with the player, like missing armor for example.

You could also try to push for some action on https://github.com/minetest/minetest/pull/4127, that should solve the problem properly.

**Comment by Cat5TV** [comment]
@Raughnbo do you want to volunteer to try replicating the issue? Stu says, Die, but NOT respawn... so just die, and let me know in IRC and I'll restart the server while you're in that dead state, and then you can try to reconnect.

**Comment by Cat5TV** [comment]
@stujones11 since upgrading, we keep crashing with: 2016-06-22 09:28:02: ERROR[Main]: ServerError: Expecting itemstack, itemstring, table or nil

**Comment by stujones11** [comment]
Is there any more to the message than that?

**Comment by Cat5TV** [comment]
Unfortunately not. That's the first sign of trouble and the server instantly goes offline.

**Comment by stujones11** [comment]
There should be more info in the logs if you revert back to the working version.

**Comment by Cat5TV** [comment]
Well, I've downgraded back to 0.4.5. Let me know what you need from me  :)

**Comment by stujones11** [comment]
I will need a more detailed traceback, there should be more information in your logs, otherwise I have no idea, it seems to be working fine for me.

**Comment by Cat5TV** [comment]
Looks like the connection is mesecons. Is something you changed affecting it?

The "new" crash happens after someone places a mesecons button.

2016-06-22 09:28:00: ACTION[Server]: xuan places node mesecons_button:button_off at (-282,2,879)
2016-06-22 09:28:00: ACTION[Server]: facedir: 2
2016-06-22 09:28:02: ACTION[Main]: IRC: Disconnected.
2016-06-22 09:28:02: ACTION[Server]: rex leaves game. List of players:
2016-06-22 09:28:02: ACTION[Server]: karhui leaves game. List of players:
2016-06-22 09:28:02: ACTION[Server]: xuan leaves game. List of players:
2016-06-22 09:28:02: ACTION[Server]: lcx leaves game. List of players:
2016-06-22 09:28:02: ACTION[Server]: dkelley810 leaves game. List of players:
2016-06-22 09:28:02: ERROR[Main]: ServerError: Expecting itemstack, itemstring, table or nil
Minetest Stopped. List of players:

For now, I'm downgraded again to 0.4.5 and things seem to run fine.

**Comment by stujones11** [comment]
Nope, neither branch has any effect on mesecons, this looks totally unrelated. Is there nothing in the logs that shows line numbers? look for `ERROR[Main]: stack traceback:` and show me what follows that if you can find it.

**Comment by Cat5TV** [comment]
What you see above is the entire error. There are no stack traceback's on this one. Minetest cacks out completely with no other notice, but only after installing 0.5.0-dev.

**Comment by stujones11** [comment]
So no warnings on start-up then? sorry, I added that bit just as you commented.

**Comment by Cat5TV** [comment]
Nothing serious.
`
2016-06-22 09:21:06: ERROR[Main]: The following mods could not be found: "no_spill_tps"
2016-06-22 09:21:07: WARNING[Main]: NodeDefManager: Ignoring CONTENT_IGNORE redefinition
2016-06-22 09:21:07: WARNING[Main]: Field "tile_images": Deprecated; new name is "tiles".
2016-06-22 09:21:07: WARNING[Main]: Field "metadata_name": Deprecated; use on_add and metadata callbacks
2016-06-22 09:21:07: WARNING[Main]: Undeclared global variable "intllib" accessed at /home/robbie/.minetest/mods/awards/init.lua:8
2016-06-22 09:21:07: WARNING[Main]: Field "maxwear" is deprecated; replace with uses=1/maxwear
2016-06-22 09:21:07: WARNING[Main]: Field "maxwear" is deprecated; replace with uses=1/maxwear
2016-06-22 09:21:07: WARNING[Main]: Field "maxwear" is deprecated; replace with uses=1/maxwear
2016-06-22 09:21:08: WARNING[Main]: Not registering alias, item with same name is already defined: stairs:stair_stone -> moreblocks:stair_stone
2016-06-22 09:21:08: WARNING[Main]: Not registering alias, item with same name is already defined: stairs:slab_stone -> moreblocks:slab_stone
2016-06-22 09:21:08: WARNING[Main]: Not registering alias, item with same name is already defined: stairs:stair_cobble -> moreblocks:stair_cobble
2016-06-22 09:21:08: WARNING[Main]: Not registering alias, item with same name is already defined: stairs:slab_cobble -> moreblocks:slab_cobble
2016-06-22 09:21:08: WARNING[Main]: Not registering alias, item with same name is already defined: stairs:stair_mossycobble -> moreblocks:stair_mossycobble
2016-06-22 09:21:08: WARNING[Main]: Not registering alias, item with same name is already defined: stairs:slab_mossycobble -> moreblocks:slab_mossycobble
2016-06-22 09:21:08: WARNING[Main]: Not registering alias, item with same name is already defined: stairs:stair_brick -> moreblocks:stair_brick
2016-06-22 09:21:08: WARNING[Main]: Not registering alias, item with same name is already defined: stairs:slab_brick -> moreblocks:slab_brick
2016-06-22 09:21:08: WARNING[Main]: Not registering alias, item with same name is already defined: stairs:stair_sandstone -> moreblocks:stair_sandstone
2016-06-22 09:21:08: WARNING[Main]: Not registering alias, item with same name is already defined: stairs:slab_sandstone -> moreblocks:slab_sandstone
2016-06-22 09:21:08: WARNING[Main]: Not registering alias, item with same name is already defined: stairs:stair_steelblock -> moreblocks:stair_steelblock
2016-06-22 09:21:08: WARNING[Main]: Not registering alias, item with same name is already defined: stairs:slab_steelblock -> moreblocks:slab_steelblock
2016-06-22 09:21:08: WARNING[Main]: Not registering alias, item with same name is already defined: stairs:stair_copperblock -> moreblocks:stair_copperblock
2016-06-22 09:21:08: WARNING[Main]: Not registering alias, item with same name is already defined: stairs:slab_copperblock -> moreblocks:slab_copperblock
2016-06-22 09:21:08: WARNING[Main]: Not registering alias, item with same name is already defined: stairs:stair_desert_stone -> moreblocks:stair_desert_stone
2016-06-22 09:21:08: WARNING[Main]: Not registering alias, item with same name is already defined: stairs:slab_desert_stone -> moreblocks:slab_desert_stone
2016-06-22 09:21:08: WARNING[Main]: Not registering alias, item with same name is already defined: stairs:stair_glass -> moreblocks:stair_glass
2016-06-22 09:21:08: WARNING[Main]: Not registering alias, item with same name is already defined: stairs:slab_glass -> moreblocks:slab_glass
2016-06-22 09:21:08: WARNING[Main]: Not registering alias, item with same name is already defined: stairs:stair_wood -> moreblocks:stair_wood
2016-06-22 09:21:08: WARNING[Main]: Not registering alias, item with same name is already defined: stairs:slab_wood -> moreblocks:slab_wood
2016-06-22 09:21:09: WARNING[Main]: Not registering alias, item with same name is already defined: stairs:stair_tar -> building_blocks:stair_tar
2016-06-22 09:21:09: WARNING[Main]: Not registering alias, item with same name is already defined: stairs:slab_tar -> building_blocks:slab_tar
2016-06-22 09:21:09: WARNING[Main]: Not registering alias, item with same name is already defined: tar -> building_blocks:Tar
2016-06-22 09:21:09: WARNING[Main]: Field "maxwear" is deprecated; replace with uses=1/maxwear
2016-06-22 09:21:09: WARNING[Main]: Field "maxwear" is deprecated; replace with uses=1/maxwear
2016-06-22 09:21:09: WARNING[Main]: Undeclared global variable "invisibility" accessed at /home/robbie/.minetest/mods/mobs_redo/api.lua:30
2016-06-22 09:21:09: WARNING[Main]: Field "noise_threshhold": Deprecated: new name is "noise_threshold".
2016-06-22 09:21:09: WARNING[Main]: Field "noise_threshhold": Deprecated: new name is "noise_threshold".
2016-06-22 09:21:09: WARNING[Main]: Field "noise_threshhold": Deprecated: new name is "noise_threshold".
2016-06-22 09:21:09: WARNING[Main]: Field "noise_threshhold": Deprecated: new name is "noise_threshold".
2016-06-22 09:21:09: WARNING[Main]: Field "noise_threshhold": Deprecated: new name is "noise_threshold".
2016-06-22 09:21:09: WARNING[Main]: Field "noise_threshhold": Deprecated: new name is "noise_threshold".
2016-06-22 09:21:09: WARNING[Main]: Field "noise_threshhold": Deprecated: new name is "noise_threshold".
2016-06-22 09:21:09: WARNING[Main]: Field "noise_threshhold": Deprecated: new name is "noise_threshold".
2016-06-22 09:21:09: WARNING[Main]: Field "noise_threshhold": Deprecated: new name is "noise_threshold".
2016-06-22 09:21:09: WARNING[Main]: Field "noise_threshhold": Deprecated: new name is "noise_threshold".
2016-06-22 09:21:09: WARNING[Main]: Undeclared global variable "gui_bg" accessed at ...me/robbie/.minetest/mods/streets/streetsmod/init.lua:32
2016-06-22 09:21:09: WARNING[Main]: Not registering alias, item with same name is already defined: stairs:stair_asphalt -> streets:stair_asphalt
2016-06-22 09:21:09: WARNING[Main]: Undeclared global variable "nb_decal" accessed at /home/robbie/.minetest/mods/soccer/init.lua:146
2016-06-22 09:21:09: WARNING[Main]: Assignment to undeclared global "texture" inside a function at /home/robbie/.minetest/mods/soccer/init.lua:155.
2016-06-22 09:21:10: WARNING[Main]: Not registering alias, item with same name is already defined: moreblocks:oerkkiblock -> gloopblocks:oerkki_block
2016-06-22 09:21:10: WARNING[Main]: Undeclared global variable "register_stair_slab_panel_mirco" accessed at ...obbie/.minetest/mods/streets/concretestairs/init.lua:4
2016-06-22 09:21:11: WARNING[Main]: Assignment to undeclared global "res" inside a function at /home/robbie/.minetest/mods/food/food/init.lua:101.
2016-06-22 09:21:13: ACTION[Main]:         .__               __                   __   
2016-06-22 09:21:13: ACTION[Main]:   _____ |__| ____   _____/  |_  ____   _______/  |_ 
2016-06-22 09:21:13: ACTION[Main]:  /     \|  |/    \_/ __ \   __\/ __ \ /  ___/\   __\
2016-06-22 09:21:13: ACTION[Main]: |  Y Y  \  |   |  \  ___/|  | \  ___/ \___ \  |  |  
2016-06-22 09:21:13: ACTION[Main]: |__|_|  /__|___|  /\___  >__|  \___  >____  > |__|  
2016-06-22 09:21:13: ACTION[Main]:       \/        \/     \/          \/     \/        `

**Comment by Cat5TV** [comment]
It stems even deeper. After putting 0.5.0-dev back, here's what happened:

> (10:03:07 AM) TPS-Survival: *** unk joined the game
(10:03:07 AM) TPS-Survival: *** unk left the game
(10:03:59 AM) TPS-Survival: *** unk joined the game
(10:03:59 AM) TPS-Survival: *** unk left the game
(10:11:25 AM) TPS-Survival: *** unk joined the game
(10:11:25 AM) TPS-Survival: *** unk left the game
(10:15:03 AM) TPS-Survival: *** unk joined the game
(10:15:03 AM) TPS-Survival: *** unk left the game
(10:16:29 AM) TPS-Creative: *** unk joined the game
(10:16:29 AM) TPS-Creative: *** unk left the game
(10:21:02 AM) TPS-Creative: *** Miranda joined the game
(10:28:01 AM) unk [~unk@2601:540:8200:284b::7bee] entered the room.
(10:28:12 AM) unk_ [~unk@2601:540:8200:284b::7bee] entered the room.
(10:28:20 AM) TPS-Creative: <Miranda> hi
(10:28:33 AM) unk_: hello
(10:28:37 AM) unk_ is now known as Guest78315
(10:29:19 AM) Guest78315: are there any problems with the pixel shadow servers today?  I am unable to connect.
(10:29:36 AM) TPS-Creative: <Miranda> nope i seem to have not even lag
(10:29:59 AM) RobbieF: Guest78315 were you banned?
(10:30:03 AM) RobbieF: what's your IP address?
(10:30:26 AM) Guest78315: this is unk...don't think so...will check for ip
(10:30:38 AM) TPS-Survival: <karhui> prospector
(10:31:30 AM) Guest78315: ip address follows:  71.206.199.131
(10:31:59 AM) RobbieF: k one sec unk
(10:32:04 AM) Guest78315: k
(10:32:06 AM) RobbieF: just to rule it out
(10:32:09 AM) RobbieF: since servers seem good
(10:32:52 AM) Guest78315: ok....able to connect to other servers and my own server on my lan
(10:32:57 AM) RobbieF: nope; ip is fine
(10:33:09 AM) RobbieF: are you on windows or linux?
(10:33:17 AM) Guest78315: ok...strange....goes to loading media and then hangs up
(10:33:18 AM) Guest78315: yes
(10:33:28 AM) RobbieF: how much ram do you have?
(10:33:32 AM) Guest78315: I cleared my cache using bleachbit as well
(10:33:34 AM) Guest78315: 10G
(10:33:37 AM) RobbieF: k
(10:33:46 AM) RobbieF: try connecting to 30004
(10:33:54 AM) Guest78315: haven't had any problems until now...will do
(10:34:07 AM) TPS-Survival: *** unk joined the game
(10:34:07 AM) TPS-Survival: *** unk left the game
(10:34:35 AM) RobbieF: which server were you having trouble with?
(10:34:35 AM) Guest78315: says i joined and left the game...but was not letting me past the load screen
(10:34:38 AM) RobbieF: oh
(10:34:40 AM) RobbieF: really?
(10:34:46 AM) RobbieF: seems a client issue then
(10:34:52 AM) RobbieF: strange that it's only us though
(10:35:02 AM) RobbieF: one sec...
(10:35:06 AM) TPS-Building: *** unk joined the game
(10:35:16 AM) RobbieF: That one works fine?
(10:35:24 AM) Guest78315: able to connect to building without issue....strange
(10:35:26 AM) RobbieF: *** SERVER RESTART *** Please reconnect in 60 seconds.
(10:35:40 AM) TPS-Creative: <Miranda> OH COME ON!
(10:35:41 AM) TPS-Building: *** unk left the game
(10:35:43 AM) TPS-Creative: *** Miranda left the game
(10:35:43 AM) RobbieF: Let's see if it's related to the 3d_armor issue.
(10:35:53 AM) Guest78315: ok
(10:35:57 AM) RobbieF: Meh, she'll thank me when she realizes how great it is to have a stable server.
(10:36:06 AM) Guest78315: :)
(10:36:19 AM) TPS-Creative left the room (quit: Remote host closed the connection).
(10:36:19 AM) TPS-Survival left the room (quit: Remote host closed the connection).
(10:36:19 AM) TPS-Realism left the room (quit: Remote host closed the connection).
(10:36:19 AM) TPS-Testing left the room (quit: Remote host closed the connection).
(10:36:19 AM) TPS-Building left the room (quit: Remote host closed the connection).
(10:36:19 AM) TPS-Xtreme left the room (quit: Remote host closed the connection).
(10:37:43 AM) TPS-Building [~Minetest@unaffiliated/thepixelshadow] entered the room.
(10:37:43 AM) mode (+v TPS-Building) by ChanServ
(10:37:44 AM) TPS-Realism [~Minetest@CPE68b6fcfa8e73-CM68b6fcfa8e70.cpe.net.cable.rogers.com] entered the room.
(10:37:45 AM) TPS-Survival [~Minetest@unaffiliated/thepixelshadow] entered the room.
(10:37:45 AM) mode (+v TPS-Survival) by ChanServ
(10:37:47 AM) TPS-Testing [~Minetest@CPE68b6fcfa8e73-CM68b6fcfa8e70.cpe.net.cable.rogers.com] entered the room.
(10:37:48 AM) TPS-Creative [~Minetest@unaffiliated/thepixelshadow] entered the room.
(10:37:48 AM) mode (+v TPS-Creative) by ChanServ
(10:37:49 AM) TPS-Xtreme [~Minetest@unaffiliated/thepixelshadow] entered the room.
(10:37:49 AM) mode (+v TPS-Xtreme) by ChanServ
(10:37:50 AM) TPS-Creative: *** Miranda joined the game
(10:37:53 AM) TPS-Survival: *** unk joined the game
(10:37:55 AM) TPS-Xtreme: *** taikedz joined the game
(10:38:09 AM) TPS-Survival: *** karhui joined the game
(10:38:11 AM) Guest78315: That worked...I am able to connect now....thanks RobbieF
(10:38:16 AM) TPS-Survival: *** lcx joined the game
(10:38:22 AM) RobbieF: k thanks unk

>

**Comment by stujones11** [comment]
That is very interested and likely relevant but it's not really giving me any clues at the moment. All I can suggest is you try again once the minetest bug gets fixed. Hopefully someone else might be able to reproduce this error in the meantime.

**Comment by Cat5TV** [comment]
Alright. Thanks stujones11. How odd eh?

Thanks for all your hard work.

Robbie

**Comment by Cat5TV** [comment]
PS - when I restarted above, that was downgrading to 0.4.5.

**Comment by stujones11** [comment]
Yeah, I guessed that it must be. I just don't understand how such a small change could cause the server to terminate like that or why a player would be unable to join, after all, you did not have this particular problem before the previous downgrade, did you?

**Comment by Cat5TV** [comment]
Correct. Before the patch, 0.5.0-dev would simply crash the server when certain players would connect, with the error that started this thread. Downgrading to 0.4.5 seems to work just fine. Then upgrading again to 0.5.0-dev after your patch caused this new error at a completely different time, and player unable to connect.

Curious; perhaps unk would have been one of the players who PREVIOUSLY would have crashed the server on join, but because of your patch, the server keeps running but that player cannot connect, and this is causing some strange issue?

**Comment by stujones11** [comment]
I think that is good possibility but without knowing where the server crashed there is little I can do to prevent it. You could try applying minetest/minetest#4127 yourself but it's a bit of a long-shot.

**Comment by stujones11** [comment]
Update. I tried that patch myself and it does not fix the original problem so don't bother wasting your time.

**Comment by Cat5TV** [comment]
Thanks for the heads up @stujones11 -- please post here if you figure it out or need me to do any testing.

**Comment by stujones11** [comment]
The 0.5.0-dev branch is now outdated against current master and will not be merged in it's current form. I will, however, be cherry-picking features from that branch so all the features will become available again eventually. Thank you for your support.
